src/Sylius/Component/Core/OrderProcessing/InventoryHandler.php

```diff
@@ -67,17 +63,17 @@ public function __construct(
      */
     public function processInventoryUnits(OrderItemInterface $item)
     {
-        $nbUnits = $item->getInventoryUnits()->count();
+        $nbUnits = $item->getUnits()->count();
 
         if ($item->getQuantity() > $nbUnits) {
             $this->createInventoryUnits($item, $item->getQuantity() - $nbUnits);
         } elseif ($item->getQuantity() < $nbUnits) {
-            foreach ($item->getInventoryUnits()->slice(0, $nbUnits - $item->getQuantity()) as $unit) {
-                $item->removeInventoryUnit($unit);
+            foreach ($item->getUnits()->slice(0, $nbUnits - $item->getQuantity()) as $unit) {
+                $item->removeUnit($unit);
             }
         }
 
-        foreach ($item->getInventoryUnits() as $unit) {
+        foreach ($item->getUnits() as $unit) {
             if ($unit->getStockable() !== $item->getVariant()) {
                 $unit->setStockable($item->getVariant());
```

Comment:
This is no longer needed, as `$unit->getStockable()` returns `$this->item->getVariant()`.